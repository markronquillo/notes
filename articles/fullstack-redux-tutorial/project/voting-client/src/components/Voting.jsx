import React from 'react';

class Voting extends React.Component {
    getPair = function() {
        return this.props.pair || [];
    }

    render = function() {
        return (
            <div className="voting">
            {this.getPair().map(entry => 
                <button key={entry}>
                    <h1>{entry}</h1>
                </button>
            )}
            </div>
        );
    }
}

export default Voting;

