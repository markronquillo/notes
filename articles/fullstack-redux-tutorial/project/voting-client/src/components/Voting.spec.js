import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Voting from '../../src/components/Voting';

describe('Voting', () => {
    
    it('renders a pair of buttons', () => {
        const component = renderer.create(
            <Voting pair={["Trainspotting", "28 Days Later"]} />
        );
        expect(component).toMatchSnapshot();
    });
})
