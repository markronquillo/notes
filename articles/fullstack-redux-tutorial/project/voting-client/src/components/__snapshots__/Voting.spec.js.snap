exports[`Voting renders a pair of buttons 1`] = `
<div
  className="voting">
  <button>
    <h1>
      Trainspotting
    </h1>
  </button>
  <button>
    <h1>
      28 Days Later
    </h1>
  </button>
</div>
`;
